core/Db/Adapter/MysqlTrait.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Db\Adapter;
+
+use Exception;
+use Piwik\Config;
+use Piwik\Piwik;
+
+/**
+ * MySQL Common
+ */
+trait MysqlTrait
+{
+    /**
+     * @var boolean
+     */
+    private $isAzure;
+
+    /**
+     * Reset the configuration variables in this adapter.
+     */
+    public function resetConfig()
+    {
+        if (isset($this->_config['host']) && strpos($this->_config['host'], '.mysql.database.azure.com') !== false) {
+            $this->isAzure = true;
+        }
+
+        $this->_config = array();
+    }
+
+    /**
+     * Return default port.
+     *
+     * @return int
+     */
+    public static function getDefaultPort()
+    {
+        return 3306;
+    }
+
+    /**
+     * Check MySQL version
+     *
+     * @throws Exception
+     */
+    public function checkServerVersion()
+    {
+        $serverVersion   = $this->getServerVersion();
+        $requiredVersion = Config::getInstance()->General['minimum_mysql_version'];
+
+        if (version_compare($serverVersion, $requiredVersion) === -1) {
+            throw new Exception(Piwik::translate('General_ExceptionDatabaseVersion', array('MySQL', $serverVersion, $requiredVersion)));
+        }
+    }
+
+    /**
+     * Returns the MySQL server version
+     *
+     * @return null|string
+     */
+    public function getServerVersion()
+    {
+        // Azure misreports the MySQL version in the connection handshake
+        if ($this->isAzure) {
+            $versionInfo = $this->fetchAll('SELECT @@VERSION');
+
+            if (count($versionInfo)) {
+                return $versionInfo[0]['@@VERSION'];
+            }
+        }
+
+        return parent::getServerVersion();
```

Comment:
Would `'SELECT @@VERSION'` work in every case? Wondering if we could just use that to avoid having to check for azure in the first place.

From what I can tell, getServerVersion() isn't used during tracking or aggregation, so I wouldn't expect there to be a performance impact.

Reply:
If it's MySQL compatible, `SELECT @@VERSION` should always work. It might look funny with MariaDB though.